analytics/src/analytics_ios.mm

```diff
@@ -312,6 +312,28 @@ void LogEvent(const char* name, const Parameter* parameters, size_t number_of_pa
   [FIRAnalytics logEventWithName:@(name) parameters:parameters_dict];
 }
 
+void SetDefaultEventParameters(const std::map<std::string, Variant>& default_parameters) {
+  FIREBASE_ASSERT_RETURN_VOID(internal::IsInitialized());
+  // Convert the std::map<std::string, Variant> to NSDictionary*
+  // The keys must be strings for FIRAnalytics.
+  NSMutableDictionary* parameters_dict =
+      [NSMutableDictionary dictionaryWithCapacity:default_parameters.size()];
+  for (const auto& pair : default_parameters) {
+    NSString* key = firebase::util::StringToNSString(pair.first);
+    // A null Variant indicates the default parameter should be cleared.
+    // In ObjC, setting a key to [NSNull null] in the dictionary achieves this.
```

Comment:
We're making the map from scratch every time so no need to delete these, just omit them.